test_examples.sh

```diff
@@ -20,25 +34,86 @@ function cleanup {
 }
 trap cleanup EXIT
 
+export ALL_OPTS=$TEMPDIR/all_opts
+if [[ "$LONG" == "--long" ]]; then
+  p() { [ $# -eq 0 ] && echo || (shift; p "$@") | while read r ; do echo -e "$1,$r\n$r"; done }
```

Comment:
I know this is just a bash script for testing infrastructure, but a one-line comment explaining what this does would be nice.

(Also, I suppose running every possible *permutation* of optimizations isn't worth it.)